packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.test.tsx

```diff
@@ -0,0 +1,527 @@
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { SelectableValue } from '@grafana/data';
+
+import { Checkbox } from '../Forms/Checkbox';
+import { RadioButtonGroup } from '../Forms/RadioButtonGroup/RadioButtonGroup';
+import { Input } from '../Input/Input';
+import { SelectBase } from '../Select/SelectBase';
+import { Switch } from '../Switch/Switch';
+import { TextArea } from '../TextArea/TextArea';
+
+import { AutoSaveField, Props } from './AutoSaveField';
+
+const mockOnFinishChange = jest.fn().mockImplementation(() => Promise.resolve());
+const mockOnFinishChangeError = jest.fn().mockImplementation(() => Promise.reject());
+const options: Array<SelectableValue<string>> = [
+  {
+    label: 'Light',
+    value: 'light',
+  },
+  {
+    label: 'Dark',
+    value: 'dark',
+  },
+  {
+    label: 'Default',
+    value: 'default',
+  },
+];
+const setup = (propOverrides?: Partial<Props>) => {
+  const props: Omit<Props, 'children'> = {
```

Comment:
in (i think) every case here you should be using `GenericProps` and passing the type argument